### PR TITLE
Adaptive timestepping callback

### PR DESCRIPTION
**Pull request by slifer50**
### Description

A simple callback to adapt the timestep to stay consistent with the user-specified initial Courant number. 
Useful for accelerating flows.

- [ ] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [ ] Unit tests are included OR N/A.
- [ ] Code is exercised in an integration test OR N/A.
- [ ] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: src/Driver/Callbacks/Callbacks.jl [hunk 1]
```diff
@@ -333,6 +334,36 @@ function monitor_courant_numbers(mcn_opt, solver_config)
...
+            dtmodel,
+            Q,
+            solver_config.CFL,
+            t0,
+            solver_config.diffdir,
+        )
+        @info @sprintf(
+            """ Updating time step: Current and former time steps %8.16f, %8.16f""",
+            ndt,
+            dt
```
Comment (kpamnany) [inline src/Driver/Callbacks/Callbacks.jl:359]:
```suggestion
            """Updating time step: %8.16f => %8.16f""",
            dt,
            ndt
```

File: src/Driver/Driver.jl [hunk 2]
```diff
@@ -218,6 +219,11 @@ function parse_commandline(
         arg_type = String
         default =
             get_setting(:monitor_courant_numbers, defaults, global_defaults)
+        "--adapt-timestep"
+        help = "interval at which to output acoustic, advective, and diffusive Courant numbers"
```
Comment (kpamnany) [inline src/Driver/Driver.jl:223]:
```suggestion
        help = "interval at which to update the timestep"
```

File: src/Driver/Driver.jl [hunk 3]
```diff
@@ -335,6 +341,8 @@ Recognized keyword arguments are:
         interval in time-steps at which to output wall-clock time per time-step
 - `monitor_courant_numbers::String = "never"`:
         interval at which to output acoustic, advective, and diffusive Courant numbers
+- `adapt-timestep::String = "never"`:
+        adapts timestep
```
Comment (kpamnany) [inline src/Driver/Driver.jl:345]:
```suggestion
        interval at which to update the timestep
```

**Comment by kpamnany** [comment]
Can add this option to the [documentation](https://github.com/CliMA/ClimateMachine.jl/blob/master/docs/src/GettingStarted/RunningClimateMachine.md) also.

**Review by kpamnany — COMMENTED** [review]
This seems like a good idea -- you have verified that it works?

I'll leave the final approval to @thomasgibson.

**Comment by slifer50** [comment]
> This seems like a good idea -- you have verified that it works?
> 
> I'll leave the final approval to @thomasgibson.

I've tested this with our time steppers and it does work, the time step is adjusted.

**Comment by slifer50** [comment]
@kpamnany Any idea what's breaking the documentation check?

**Comment by slifer50** [comment]
bors try

**Review by thomasgibson — APPROVED** [review]

**Comment by slifer50** [comment]
@kpamnany can we r+ this?

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
(Just killing the try since it's already been r+'d)

**Comment by jkozdon** [comment]
bors try-

**Comment by jkozdon** [comment]
bors try-

**Comment by kpamnany** [comment]
> (Just killing the try since it's already been r+'d)

I've never been able to `try-`. 🤷‍♂️

**Comment by jkozdon** [comment]
> I've never been able to `try-`. 🤷‍♂️

It should work.

    https://bors.tech/documentation/

Fortunately, bors seems to have gotten out of it's stuck state after I try- my PR and re-tried it.

Technologies great until it fails 😞

**Pull request merged**